test/functional/wallet_orphanedreward.py

```diff
@@ -32,28 +32,31 @@ def run_test(self):
         self.generate(self.nodes[0], 150)
         assert_equal(self.nodes[1].getbalance(), 10 + 25)
         txid = self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 30)
+        orig_chain_tip = self.nodes[0].getbestblockhash()
 
         # Orphan the block reward and make sure that the original coins
         # from the wallet can still be spent.
         self.nodes[0].invalidateblock(blk)
-        self.generate(self.nodes[0], 152)
-        # Without the following abandontransaction call, the coins are
-        # not considered available yet.
-        assert_equal(self.nodes[1].getbalances()["mine"], {
-          "trusted": 0,
-          "untrusted_pending": 0,
-          "immature": 0,
-        })
-        # The following abandontransaction is necessary to make the later
-        # lines succeed, and probably should not be needed; see
-        # https://github.com/bitcoin/bitcoin/issues/14148.
-        self.nodes[1].abandontransaction(txid)
+        blocks = self.generate(self.nodes[0], 152)
+        conflict_block = blocks[0]
+        # We expect the desendants of orphaned rewards to no longer be considered
         assert_equal(self.nodes[1].getbalances()["mine"], {
           "trusted": 10,
           "untrusted_pending": 0,
           "immature": 0,
         })
-        self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 9)
+        # And the unconfirmed tx to be abandoned
+        assert_equal(self.nodes[1].gettransaction(txid)["details"][0]["abandoned"], True)
+
+        # If the orphaned reward is reorged back into the main chain, any unconfirmed txs
+        # at the time of the original reorg remain abandoned.
+        self.nodes[0].invalidateblock(conflict_block)
+        self.nodes[0].reconsiderblock(blk)
+        assert_equal(self.nodes[0].getbestblockhash(), orig_chain_tip)
+        self.generate(self.nodes[0], 3)
```

Comment:
Should we let the block reward mature again?? Then show that the reward is included in the balance but the descendant is still abandoned?  I manually verified that's the case..